patches/minecraft/net/minecraft/util/math/BlockPos.java.patch

```diff
@@ -1,41 +1,10 @@
 --- ../src-base/minecraft/net/minecraft/util/math/BlockPos.java
 +++ ../src-work/minecraft/net/minecraft/util/math/BlockPos.java
-@@ -270,6 +270,21 @@
-         };
-     }
- 
-+    /**
-+     * Returns a version of this BlockPos that is guaranteed to be Immutable.
-+     * In most cases this will return 'this', but if 'this' is an instance of
-+     * MutableBlockPos it will return a new instance of BlockPos with the same values.
-+     *
-+     * When storing a parameter given to you for an extended period of time, make sure you
-+     * use this in case the value is changed internally.
-+     *
-+     * @return An immutable BlockPos.
-+     */
-+    public BlockPos getImmutable()
-+    {
-+        return this;
-+    }
-+
-     public static final class MutableBlockPos extends BlockPos
-         {
-             private int field_177997_b;
-@@ -333,6 +348,8 @@
-             {
-                 return new BlockPos(this);
-             }
-+
-+            @Override public BlockPos getImmutable() { return new BlockPos(this); }
-         }
- 
-     public static final class PooledMutableBlockPos extends BlockPos
-@@ -443,5 +460,7 @@
+@@ -443,5 +443,7 @@
              {
                  return this.func_185343_d(this.field_185347_c + p_185341_1_.func_82601_c(), this.field_185348_d + p_185341_1_.func_96559_d(), this.field_185349_e + p_185341_1_.func_82599_e());
              }
 +
-+            @Override public BlockPos getImmutable() { return new BlockPos(this); }
```

Comment:
`PooledMutableBlockPos` does not override the `toImmutable` method in vanilla, so we need to in Forge to prevent a mutable version being passed around. MC-100165